packages/core/src/extensions/Blocks/nodes/BlockContainer.ts

```diff
@@ -253,8 +253,13 @@ export const BlockContainer = Node.create<{
                     content
                   )
                 )
+                // If the node doesn't contain editable content, we want to
+                // select the whole node. But if it does have editable content,
+                // we want to set the selection to the start of it.
                 .setSelection(
-                  new NodeSelection(state.tr.doc.resolve(startPos))
+                  state.schema.nodes[newType].spec.content === ""
+                    ? new NodeSelection(state.tr.doc.resolve(startPos))
+                    : new TextSelection(state.tr.doc.resolve(startPos))
```

Comment:
why don't we need the offset logic ( + 4 or + 1) for textselection here, but we do need it in setTextCursorPosition?

Reply:
It does actually throw a warning with +1, i guess PM just shifts it to the nearest inline content but it should also be +4